### PR TITLE
Qos demo options

### PR DESCRIPTION
**Pull request by esteve**
Adds command line parsing to `cam2image` and `showimage`.

A future PR will add support for passing XML files with QoS profiles.

### CODE REVIEW

**Comment by esteve** [comment]
CI jobs:

http://ci.ros2.org/job/ros2_batch_ci_linux/186/
http://ci.ros2.org/job/ros2_batch_ci_osx/116/

**Comment by dirk-thomas** [comment]
http://ci.ros2.org/job/ros2_batch_ci_windows/162/

The OS X link seems to be wrong.

**Comment by wjwwood** [comment]
+1

**Comment by esteve** [comment]
I started another CI job on the OS X slave:

http://ci.ros2.org/job/ros2_batch_ci_osx/118/

**Comment by jacquelinekay** [comment]
Uncrustify shows a few style errors.

**Comment by esteve** [comment]
The CI job on Linux is green, I don't seen any uncrustify errors:

http://ci.ros2.org/job/ros2_batch_ci_linux/211/

**Comment by jacquelinekay** [comment]
Strange, maybe my local install of uncrustify is wrong or overly picky.

File: image_tools/src/showimage.cpp [hunk 1]
```diff
@@ -21,13 +21,14 @@
...
 
+#include <image_tools/options.hpp>
 
 int
 encoding2mat_type(const std::string & encoding)
 {
   if(encoding == "mono8") {
     return CV_8UC1;
-  } else if(encoding == "rgb8") {
+  } else if(encoding == "bgr8") {
```
Comment (jacquelinekay) [inline image_tools/src/showimage.cpp]: Uncrustify complains about no space before the paren in `if`

Comment (esteve) [inline image_tools/src/showimage.cpp]: I added a space between the if and the paren and started a new CI job (waiting for the previous one to finish):

http://ci.ros2.org/job/ros2_batch_ci_linux/215/ (not started yet)

**Comment by dirk-thomas** [comment]
Please fix the new warnings on Windows: http://ci.ros2.org/job/ros2_batch_ci_windows/221/warnings34Result/new/

**Comment by esteve** [comment]
Pushed a change to `qos-demo-options-2` to remove the warnings, it's being built here:

http://ci.ros2.org/job/ros2_batch_ci_windows/223/

**Comment by esteve** [comment]
The warnings are gone, but the build is not green. The tests were already broken before pushing this.

**Comment by dirk-thomas** [comment]
The remaining 10 warnings have been there for a long time and the failing tests also. So good to merge.

**Pull request merged**